terraphim_server/tests/server.rs

```diff
@@ -0,0 +1,184 @@
+//! Integration tests for the server
+//!
+//! These tests are meant to be run against a running server.
+//! We test the server by sending requests to it and checking the responses.
+#[cfg(test)]
+mod tests {
+    use terraphim_server::axum_server;
+    use terraphim_settings::Settings;
+
+    use reqwest::{Client, StatusCode};
+    use std::net::SocketAddr;
+    use terraphim_config::{ConfigState, ServiceType, TerraphimConfig};
+    use tokio::sync::OnceCell;
+
+    static SERVER: OnceCell<()> = OnceCell::const_new();
+
+    async fn start_server() {
+        let server_settings =
+            Settings::load_from_env_and_file(None).expect("Failed to load settings");
+        let server_hostname = server_settings
+            .server_hostname
+            .parse::<SocketAddr>()
+            .unwrap_or_else(|_| {
+                let port = portpicker::pick_unused_port().expect("Failed to find unused port");
+                SocketAddr::from(([127, 0, 0, 1], port))
+            });
+
+        let mut config = TerraphimConfig::new(ServiceType::Ripgrep);
+        let config_state = ConfigState::new(&mut config)
+            .await
+            .expect("Failed to create config state");
+
```

Comment:
@mre here should be code for system operator role: 
```
    // Example of adding a role for testing
    let role = "system operator2".to_string();
    let automata_url = "https://system-operator.s3.eu-west-2.amazonaws.com/term_to_id.json";
    let rolegraph = RoleGraph::new(role.clone(), automata_url).await?;
```

Reply:
As discussed, the default `TerraphimConfig` sets up a system operator role. The tests pass in their current state, but in order to make them correct we need to check the response body, and for that more work needs to be done. (Like setting up the `/tmp/system_operator/pages/` path before the test.)

Same for the older pull request #46. IMHO, this can be done in a separate PR.